airgap-infra/docker_registry.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+root_dir="$( cd "$( dirname $0 )" && pwd )"
+port=5000
+data_dir=/data
+cert_dir=/certs
+tls_cert=
+tls_key=
+
+
+installDocker(){
+    curl -fsSL https://get.docker.com/ | sh
+    service docker start
+}
+
+
+pullRegistry(){
+    docker pull registry:2
+
+    mkdir -p data_dir
+    mkdir -p cert_dir
+
+    docker run -d -p $port:5000 -v $data_dir:/var/lib/registry registry:2
+}
+
+download_images() {
+
+IMAGES_CACHE=/tmp/kubespray_cache
+INVENTORY_FILE=${root_dir}/air-gap.ini
+
+mkdir -p ${IMAGES_CACHE}/images
+
+mkdir -p /etc/kubernetes
+
+ansible-playbook -i  ${INVENTORY_FILE}  ${root_dir}/../cluster.yml \
+	--tags download  \
+	-e download_run_once=true \
+	-e download_localhost=false \
+	--become --become-user root
+        -e download_cache_dir="${IMAGES_CACHE}" \
+        -e proxy_env='{"HTTPS_PROXY": "", "HTTP_PROXY": "", "NO_PROXY": "", "http_proxy": "", "https_proxy": "", "no_proxy": ""}' \
+        -e ansible_architecture="x86_64" \
+        -e ansible_system="Linux" \
+        -e ansible_os_family="RedHat" 
+
+}
+
+loadPublicImages(){
+    public_img=$(docker images | awk '{print $1":" $2}' | grep -v REPO)
+    for s in $public_img; do
+        #docker pull $s
+        docker tag $s localhost:5000/$s
```

Comment:
I guess this way cannot cover `k8s.gcr.io/addon-resizer:1.8.11` image.
Most images are loaded without containing registry names(`k8s.gcr.io`, `docker.io`, etc.) on docker engine, and it is possible to replace these registry names on Kubespray with `localhost:5000` by specifying the options like `kube_image_repo` , `docker_image_repo` and so on.
However, addon-resizer image is loaded by including `k8s.gcr.io` in the name like `k8s.gcr.io/addon-resizer:1.8.11` and here can be `localost:5000/k8s.gcr.io/addon-resizer:1.8.11`.
Then a difference between loaded image names and Kubespray specified names happens and pulling images will be failed I guess.
I did the similar implementation as https://github.com/kubernetes-sigs/kubespray/pull/7024 and I just removes the repository names from images to avoid this issue.

Can here avoid this issue?